plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1769,6 +1775,420 @@ public void testSupportedNonPartitionedColumnMappingNoneWrites()
+            assertTableType("test_schema", table.getName(), EXTERNAL_TABLE.name());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " (colA VARCHAR)");
+            assertTableType("test_schema", table.getName(), EXTERNAL_TABLE.name());
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableAsWithNoLocationSpecifiedForExternalTable()
+    {
+        String location = "s3://%s/%s".formatted(bucketName, randomNameSuffix());
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "create_or_replace_with_no_location_",
+                " (a BIGINT) WITH (location = '%s')".formatted(location))) {
+            assertTableType("test_schema", table.getName(), EXTERNAL_TABLE.name());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT 'abc' as colA", 1);
+            assertTableType("test_schema", table.getName(), EXTERNAL_TABLE.name());
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWithNoLocationSpecifiedForManagedTable()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "create_or_replace_with_no_location_",
+                " (a BIGINT)")) {
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " (colA VARCHAR)");
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableAsWithNoLocationSpecifiedForManagedTable()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "create_or_replace_with_no_location_",
+                " (a BIGINT)")) {
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT 'abc' as colA", 1);
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWithStatsUpdated()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "create_or_replace_for_stats_",
+                " AS SELECT 1 as colA")) {
+            assertQuery(
+                    "SHOW STATS FOR " + table.getName(),
+                    "VALUES" +
+                            "('cola', null, 1.0, 0.0, null, '1', '1')," +
+                            "(null, null, null, null, 1.0, null, null)");
+
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " (colA BIGINT) ");
+            assertQuery(
+                    "SHOW STATS FOR " + table.getName(),
+                    "VALUES" +
+                            "('cola', 0.0, 0.0, 1.0, null, null, null)," +
```

Comment:
Why is the `nulls_fraction` set to `1` here?
I know it is an estimation, but don't follow how it got `1` when the table is empty.

Reply:
The same is shown for `CREATE TABLE` as well - I guess it is as a part of https://github.com/trinodb/trino/blob/0bd706234d62736222f6417ae4c20b449de52195/core/trino-main/src/main/java/io/trino/cost/SymbolStatsEstimate.java#L34